head of personnel's hardsuit

SomeCabbage opened the pull request:
# Description
Changes to the parkstation hardsuit.yml to add HoP's hardsuit
Changes to the parkstation hardsuit-helmets.yml to add HoP's hardsuit helmet
Sprites added for the entirety of the HoP's hardsuit and the helmet
Command encryption key box added
Command encryption key box and command remote added to HoP's locker

# TODO
- [x] make the hardsuit
- [ ] Change the HoP's locker in the map to be the filled catalog one
- [ ] get the hardsuit added
- [x] integrate it into locker spawns

# Media
<!--
If applicable, add screenshots or videos to showcase your PR.
This makes it much easier for us to merge PRs.
Refactors or small fixes are exempt, but all PRs which make ingame changes (adding clothing, items, new features, etc) must include ingame media or the PR will not be merged.
Use screenshot software like Window's built in snipping tool, ShareX, Lightshot, or recording software like ShareX (gif), ScreenToGif, or Open Broadcaster Software (cross platform).
If you're unsure whether your PR will require media, ask a maintainer.
-->
![hopsuit1](https://user-images.githubusercontent.com/130223709/231056491-5d0804dc-2785-4399-ae63-d2cc13b40ba6.PNG)
![hopsnip2](https://user-images.githubusercontent.com/130223709/231056494-fbc0735c-5267-47fd-9638-230c1f699aaa.PNG)
![hopsnip3](https://user-images.githubusercontent.com/130223709/231056496-05fcfb81-1f59-4462-99a0-121d1c53a46d.PNG)

# Changelog
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged.
There are 4 icons for changelog entries: add, remove, tweak, fix.

You can put your name after the :cl: symbol to change the name that shows in the changelog.
Like so:
:cl: PJB


Generally, only put things in changelogs that players actually care about.
Stuff like "Refactored X system, no changes should be visible" shouldn't be on a changelog.

For writing actual entries, don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers
-->

:cl: Cabbage
- add: Added head of personnel's plated hardsuit.

SomeCabbage committed:
Changed hardsuit-helmets.yml and hardsuits.yml to include the HoP hardsuit and it's image assets

SomeCabbage committed:
adjusted damage values for the hardsuit

SomeCabbage committed:
modified art assets for the hardsuit

SomeCabbage committed:
changed the sprites again

SomeCabbage committed:
changed art sprites again

DEATHB4DEFEAT commented:
Could put it in the HoP locker instead of mapping it

SomeCabbage commented:
is there a list of items that are put into the lockers?

DEATHB4DEFEAT commented:
I believe `Resources/Prototypes/Catalog/Fills/Lockers/heads.yml`

Pspritechologist commented on Resources/Textures/SimpleStation14/Clothing/Head/Hardsuits/hopsuit.rsi/meta.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "version": 1,
+  "license": "",
+  "copyright": "",
```
These need to include your license, and copyright information. Since these are new sprites by you, license can be anything (Just use one of the default 14 ones, you'll find it in most files), and the copyright should just credit you.

Pspritechologist commented on Resources/Textures/SimpleStation14/Clothing/OuterClothing/Hardsuits/hopsuit.rsi/meta.json:
```diff
@@ -0,0 +1,26 @@
+{
+  "version": 1,
+  "license": "",
+  "copyright": "",
```
DItto.

Pspritechologist commented on Resources/Prototypes/SimpleStation14/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -13,3 +13,27 @@
+    color: "#ff2034"
+  - type: PressureProtection
+    highPressureMultiplier: 0.525
+    lowPressureMultiplier: 10000
```
```suggestion
    lowPressureMultiplier: 1000
```

Hardsuit helmet values are _all_ over the place, but 1000 seems to be normal for non-combat ones.

SomeCabbage replied:
holy fuck they are. I have no clue what really is a standard

Pspritechologist commented on Resources/Prototypes/SimpleStation14/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -13,3 +13,27 @@
+        Slash: 0.95
+        Piercing: 0.95
+        Heat: 0.9
+        Radiation: 0.9
```
```suggestion
        Radiation: 0.6
```
Even a _bad_ hardsuit should probably still be really good for radiation :P

Pspritechologist commented on Resources/Prototypes/SimpleStation14/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -27,3 +27,31 @@
+        Slash: 0.8
+        Piercing: 0.90
+        Heat: 0.8
+        Radiation: 0.7
```
```suggestion
        Radiation: 0.6
```

Pspritechologist commented on Resources/Prototypes/SimpleStation14/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -27,3 +27,31 @@
+    sprite: SimpleStation14/Clothing/OuterClothing/Hardsuits/hopsuit.rsi
+  - type: PressureProtection
+    highPressureMultiplier: 0.02
+    lowPressureMultiplier: 10000
```
```suggestion
    lowPressureMultiplier: 4500
```

5500 Seems to be about standard for non-combat hardsuits, so I'd think this would make sense for HoP.

Pspritechologist commented on Resources/Prototypes/SimpleStation14/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -27,3 +27,31 @@
+- type: entity
+  parent: ClothingOuterHardsuitBase
+  id: ClothingOuterHardsuitHoP
+  name: head of personnel's plated hardsuit
```
"Plated" somewhat implies fortified, which is a little misleading.

Pspritechologist commented on Resources/Textures/SimpleStation14/Clothing/Head/Hardsuits/hopsuit.rsi/on-equipped-HELMET.png:
The way lights and the like are done is with overlays, which means the actual helmet itself doesn't need to be in this sprite, just the light that goes overtop of it. @ me in Discord when you get to this, as I have a useful tip for dealing with this.

SomeCabbage committed:
Update Resources/Prototypes/SimpleStation14/Entities/Clothing/Head/hardsuit-helmets.yml

accepted suggestion cuz it was good

Co-authored-by: Pspritechologist <81725545+Pspritechologist@users.noreply.github.com>

SomeCabbage committed:
set the suit to spawn in hop's locker
increased and readjusted resistance values and suit name to avoid confusion.
updated copyright and license

DEATHB4DEFEAT commented on Resources/Prototypes/SimpleStation14/Catalog/Fills/Lockers/heads.yml:
```diff
@@ -12,3 +12,30 @@
       - id: PenCSS
         prob: 0.4
+
+- type: entity
+  id: LockerHeadOfPersonnelFilled
+  suffix: Filled
+  parent: LockerHeadOfPersonnel
+  components:
+  - type: StorageFill
+    contents:
+      - id: ClothingNeckCloakHop
+      - id: ClothingHeadHatHopcap
+      - id: HoPIDCard
+        prob: 0.9
+      - id: ClothingHeadsetAltCommand
+      - id: BoxPDA
+      - id: BoxID
+      - id: BoxHeadset
+      - id: ClothingShoesColorBlack
+        prob: 0.7
+      - id: PlushieLizard
+        prob: 0.1
+      - id: CigarGoldCase
+        prob: 0.10
+      - id: ClothingNeckGoldmedal
+      - id: RubberStampHop
+      - id: BoxEncryptionKeyPassenger
+      - id: BoxEncryptionKeyService
+      - id: ClothingOuterHardsuitHoP
```
if there was already a filled variation in the upstream files, did you remove it? The modifications should probably just be in the original file instead of our namespace

SomeCabbage replied:
There was an original in the upstream but I was told to not typically mess with upstream stuff. So, I'm not too sure.

Pspritechologist replied:
That's understandable. The line is somewhat blurry, but generally, if we're tweaking Upstream things we just tweak them. If we're adding new content, that goes in our files.

SomeCabbage replied:
changed it

DEATHB4DEFEAT reviewed (commented):
This still a draft?

SomeCabbage committed:
added a bit more chunkiness to the suit sprite

SomeCabbage committed:
changed light sprite for hop helm

SomeCabbage committed:
ditto

SomeCabbage committed:
redid the heads of personnel locker files and changed the upstream on instead of adding to the branch

SomeCabbage commented:
like uh what do I do I dunno what to do how do I advance this process

DEATHB4DEFEAT commented:
Random comment from someone in the Nyanotrasen Discord I thought I'd share
![image](https://img.simplemodbot.tk/Kp49wC45L4Vc.png)
![image](https://img.simplemodbot.tk/fN5e1WRTeWcr.png)
![image](https://img.simplemodbot.tk/mTsAVZwHJHlc.png)
![image](https://img.simplemodbot.tk/1xpcjftS9O2m.png)
![image](https://img.simplemodbot.tk/ZpjijUao3Tkr.png)

SomeCabbage committed:
redid the sprites and how they wokred

SomeCabbage committed:
oops forgot to change the directory for the hopsuit

SomeCabbage committed:
changes again to sprites and metadata

SomeCabbage committed:
forgot a comma

SomeCabbage commented:
Okay so I really did change the way the hardsuit helmet sprites works and now it uses the same system as the atmos hardsuit. I decided I didn't want glowy bits so I omitted that.

DEATHB4DEFEAT commented on Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:
```diff
@@ -55,6 +55,7 @@
       - id: WeaponAntiqueLaser
       - id: JetpackCaptainFilled
 
+
```
```suggestion
```

DEATHB4DEFEAT commented on Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:
```diff
@@ -65,24 +66,19 @@
-    #  - id: DoorRemoteService
       - id: ClothingNeckGoldmedal
       - id: RubberStampHop
-      - id: BoxEncryptionKeyPassenger
```
muh encryption keys

SomeCabbage committed:
Added command encrytion key box. Added science encrytion key box to MG's locker, added command encryption key box to HoP's locker

SomeCabbage committed:
added back common encryption keys to hop's locker cuz idk it might have to be done

SomeCabbage committed:
Update Resources/Prototypes/Catalog/Fills/Lockers/heads.yml

yep

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>

SomeCabbage reviewed (commented):
test uh i dunno what to put here

Pspritechologist commented on Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:
```diff
@@ -65,24 +65,21 @@
       - id: CigarGoldCase
         prob: 0.10
-        # Fuck the HoP they don't deserve fucking cigars.
-    #  - id: DoorRemoteService
```
I probably should've done this quite some time ago, but change this to DoorRemoteCommand and add it back.

SomeCabbage replied:
righto, it's done

Pspritechologist commented on Resources/Prototypes/Nyanotrasen/Catalog/Fills/Lockers/ourjobs.yml:
```diff
@@ -13,6 +13,7 @@
       - id: DoorRemoteResearch
       - id: HandTeleporter
       - id: RubberStampRd
+      - id: BoxEncryptionKeyScience
```
Changing little things like this is usually best left to the Upstream that made it, as to not cause conflicts when pulling. I imagine Nyano has already gotten to this in the updates we haven't merged.
That said, I'm fine with leaving it for now. Just thought I'd let you know for future.

SomeCabbage replied:
I'll make sure to keep that in mind. I'll revert it for now. (just in case)

Pspritechologist reviewed (changes requested):
Couple of things I mentioned, looks good other than that! Nice job getting the lights to work properly, the difference in game really will be noticeable.

SomeCabbage committed:
added command remote to locker, reverted the MG locker to the way it was before

Pspritechologist commented:
Now the CSS needs one! :P
